### PR TITLE
Ensure WP image URL check support URL encoded paths

### PR DESCRIPTION
**Pull request by roborourke**
As of WP 5.5 a new function `wp_image_file_matches_image_meta()` is used to check that a given image URL corresponds to the provided image metadata during inline editing. This function does not account for URL encoded image sources which is how Smart Media stores images in content.

Fixes https://github.com/humanmade/altis-media/issues/91

### CODE REVIEW

**Review by jazzsequence — CHANGES_REQUESTED** [review]
Pulled this down locally and tested a couple times. I mostly get the same error (see screencap), however once it took a very long time and I got an "Invalid JSON response" (or similar) error, but I was unable to reproduce that again.

https://www.dropbox.com/s/p3gtmguhzvxq82y/altis-v5-qa-fix-wp-image-size-check.mov?dl=0

File: inc/cropper/namespace.php [hunk 1]
```diff
@@ -1254,3 +1257,42 @@ function editor_max_image_size( array $size_array, $size ) : array {
...
+ * Filter wp_image_file_matches_image_meta() to add support for URL encoded images.
+ *
+ * In some cases Tachyon or smart media can return or store a URL encoded image src.
+ * WordPress checks the passed image URL against the image metadata as a safety check
+ * when editing images inline but it does not account for URL encoded paths.
+ *
+ * @param bool $match Whether the image has matched or not.
+ * @param string $image_location The image URL to compare.
+ * @param array $image_meta Attachment meta data.
+ * @param int $attachment_id The attachment ID.
```
[the remark this answers is not in the document]
Comment (roborourke) [reply inc/cropper/namespace.php:1271]: We no longer require this in our coding standards, for the most part the docblocks are consumed by your editor to show the arguments and inline docs as you type - it's not that often you actually need to look at the docblock directly.

Other than that this approach avoids code churn - say you add a new param that's quite long, you have to realign all the other params even though they haven't actually changed.

File: inc/cropper/namespace.php [hunk 2]
```diff
@@ -1254,3 +1257,42 @@ function editor_max_image_size( array $size_array, $size ) : array {
...
+	}
+
+	// URL decode the image src.
+	$image_location = urldecode( $image_location );
+
+	// Add a flag to image meta to avoid recursion.
+	$image_meta['is_url_decoded'] = true;
+
+	// Check again for a match.
+	return wp_image_file_matches_image_meta( $image_location, $image_meta, $attachment_id );
```
[the remark this answers is not in the document]
Comment (roborourke) [reply inc/cropper/namespace.php:1297]: A little bit - it'll actually just compare the image file name in `$image_meta['file']` with the URL in `$image_location` and return early in that function. I'd say it's a pretty minimal hit to make sure everything works.

File: inc/cropper/namespace.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (jazzsequence) [reply inc/cropper/namespace.php:1271]: Fair enough. Still bugs me. 😜

**Comment by jazzsequence** [comment]
Was able to confirm that this is working as expected when I'm testing the right thing 😅

**Review by jazzsequence — APPROVED** [review]

**Pull request merged**